File: internal/content/testdata/bench/typescript/product-of-letter-counts.ts
function product_of_counts(s: string): number {
    const MOD = 1000000007;
    const freq: {[key: string]: number} = {};
    for (const ch of s) {
        freq[ch] = (freq[ch] || 0) + 1;
    }
    let result = 1;
    for (const cnt of Object.values(freq)) {
        result = (result * cnt) % MOD;
    }
    return result;
}

File: internal/content/testdata/bench/java/product-of-letter-counts.java
import java.util.*;
class Solution {
    public long product_of_counts(String s) {
        long MOD = 1_000_000_007L;
        Map<Character, Long> freq = new HashMap<>();
        for (char ch : s.toCharArray()) {
            freq.put(ch, freq.getOrDefault(ch, 0L) + 1);
        }
        long result = 1;
        for (long cnt : freq.values()) {
            result = (result * cnt) % MOD;
        }
        return result;
    }
}
